pkg/util/helper/work.go

```diff
@@ -117,15 +118,19 @@ func GetWorksByLabelsSet(ctx context.Context, c client.Client, ls labels.Set) (*
 }
 
 // GetWorksByBindingID gets WorkList by matching same binding's permanent id.
+// Caller should ensure Work is indexed by binding's permanent id.
 func GetWorksByBindingID(ctx context.Context, c client.Client, bindingID string, namespaced bool) (*workv1alpha1.WorkList, error) {
-	var ls labels.Set
+	var key string
 	if namespaced {
-		ls = labels.Set{workv1alpha2.ResourceBindingPermanentIDLabel: bindingID}
+		key = workv1alpha2.ResourceBindingPermanentIDLabel
 	} else {
-		ls = labels.Set{workv1alpha2.ClusterResourceBindingPermanentIDLabel: bindingID}
+		key = workv1alpha2.ClusterResourceBindingPermanentIDLabel
 	}
-
-	return GetWorksByLabelsSet(ctx, c, ls)
+	workList := &workv1alpha1.WorkList{}
+	listOpt := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(key, bindingID),
+	}
+	return workList, c.List(ctx, workList, listOpt)
```

Comment:
I believe this can significantly accelerate the query efficiency, as it doesn't need to query all Works.

I'm thinking about how to make a benchmark test for this, I'm curious how much performance it enhanced based on the upstream codebase. Any thoughts?

Reply:
One thing for sure is, if the test result is triggered by CI, then we must add the test before this commit.

Reply:
I think I found a way to benchmark this part without using fakeClient or network requests.

The `cache.Cache` from controleller-runtime could be either `client.Reader` and `client.FieldIndexer`, so we could use this cache as a resource provider to pass into the function. But we may need to change the signature of function `GetWorksByBindingID()`, replace `c client.Client` with `c client.Reader`.  For the benchmark without index, we can only write another function.

As a scenario with different performances when the amount of resources is different, I think the amount of resources can be used as the input parameter of the benchmark. This is different from the common benchmark, but I think it will be more effective in showing performance gaps.

Reply:
Updated. I finally chose to use the envtest package in controller-runtime to start the local apiserver, to implement the benchmark.

kindly ping @RainbowMango to review again.

![image](https://github.com/user-attachments/assets/0757fda5-c829-4891-81ad-cdb990d8b07a)